File: emarsys-sdk/src/main/java/com/emarsys/config/ConfigApi.java
package com.emarsys.config;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.emarsys.core.api.result.CompletionListener;

public interface ConfigApi {

    @NonNull
    int getContactFieldId();

    void changeApplicationCode(@Nullable String applicationCode, @Nullable CompletionListener completionListener);

    @Nullable
    String getApplicationCode();

    void changeMerchantId(@Nullable String merchantId);
}

File: emarsys-sdk/src/main/java/com/emarsys/config/ConfigApi.kt
package com.emarsys.config

import com.emarsys.core.api.notification.NotificationSettings
import com.emarsys.core.api.result.CompletionListener

interface ConfigApi {

    val contactFieldId: Int?

    val applicationCode: String?

    val merchantId: String?

    val hardwareId: String

    val languageCode: String

    val notificationSettings: NotificationSettings

    val isAutomaticPushSendingEnabled: Boolean

    val sdkVersion: String

    fun changeApplicationCode(applicationCode: String?)

    fun changeApplicationCode(applicationCode: String?, completionListener: CompletionListener?)

    fun changeMerchantId(merchantId: String?)
}
